cmd/wallet-web/src/components/Vaults/VaultCard.vue

```diff
@@ -1,67 +1,65 @@
-        >
-          <img class="w-6 h-5" src="@/assets/img/vaults.svg" alt="Vault Icon" />
-        </div>
-      </div>
-      <div>
-        <div class="px-3 md:px-0 pb-4">
-          <span class="block pt-4 text-lg font-bold text-neutrals-dark"> {{ name }}</span>
-          <span class="block text-sm text-neutrals-medium">
-            {{ numOfCreds }}
-          </span>
-        </div>
+  <router-link
+    :to="{ name: 'credentials', params: { vaultId } }"
+    v-if="type === 'regular'"
+    class="
+      md:flex-col
+      px-6
+      w-full
+      xl:w-64
+      h-20
+      md:h-auto
+      bg-neutrals-white
+      rounded-xl
+      flex flex-row
+    "
+  >
+    <div class="pt-5">
+      <div
+        :class="[`flex justify-center items-center w-12 h-12 bg-gradient-${color} rounded-full`]"
+      >
+        <img class="w-6 h-5" src="@/assets/img/vaults.svg" alt="Vault Icon" />
       </div>
     </div>
-    <div
-      v-if="type === 'addNew'"
-      class="
-        px-6
-        w-full
-        xl:w-64
-        h-24
-        md:h-full
-        bg-neutrals-moist
-        rounded-xl
-        border border-neutrals-dark border-opacity-10
-        flex flex-col
-      "
-    >
-      <div class="flex justify-center pt-5 md:pt-10">
-        <div class="flex justify-center items-center w-8 h-8 bg-neutrals-white rounded-full">
-          <img
-            class="w-6 h-5 text-primary-purple"
-            src="@/assets/img/icons-sm--plus-icon.svg"
-            alt="Add Icon"
-            @click="showAddVault = !showAddVault"
-          />
-        </div>
-      </div>
-      <div class="flex justify-center">
-        <span class="block pt-2 pb-4 text-base font-bold text-neutrals-dark">
-          {{ t('Vaults.addVault') }}</span
-        >
+    <div>
+      <div class="px-3 md:px-0 pb-4">
+        <span class="block pt-4 text-lg font-bold text-neutrals-dark"> {{ name }}</span>
+        <span class="block text-sm text-neutrals-medium">
+          {{ numOfCreds }}
+        </span>
       </div>
     </div>
-    <add-vault v-if="showAddVault" @close="showAddVault = false" />
-  </div>
+  </router-link>
+  <button
+    v-if="type === 'addNew'"
+    class="
+      px-6
+      w-full
+      xl:w-64
+      h-24
+      md:h-full
+      bg-neutrals-moist
+      rounded-xl
+      border border-neutrals-dark border-opacity-10
+      flex flex-col
+      justify-center
+      items-center
+      md:pt-5
+    "
+    @click="showAddVault = !showAddVault"
+  >
+    <div class="w-8 h-8 bg-neutrals-white rounded-full flex justify-center items-center">
+      <img
+        class="w-6 h-5 text-primary-purple"
+        src="@/assets/img/icons-sm--plus-icon.svg"
+        alt="Add Icon"
+      />
+    </div>
+    <span class="block pt-2 pb-4 text-base font-bold text-neutrals-dark">
+      {{ t('Vaults.addVault') }}</span
+    >
+  </button>
+  <add-vault v-if="showAddVault" @close="showAddVault = false" />
```

Comment:
And this modal should only be rendered in case both `type === 'addNew'` and `showAddVault` is true, so we should probably move it inside the button

Reply:
no , i tested earlier as well it doesnt work properly, as if i put modal inside button, that button click event just closes the modal as well , as soon you try put some input